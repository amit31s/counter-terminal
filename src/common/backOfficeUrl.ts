import { STORAGE_KEYS } from "@ct/common/enums";
import { getItem } from "@ct/utils";
import { envProvider } from "./platformHelper";

export enum BackOfficePageEnum {
  taskManagement = "taskManagement",
  pouchPreperation = "pouchPreperation",
}

export const generateBackOfficeURL = () => {
  const currentPage = getItem(STORAGE_KEYS.CTSTK0003);
  const deviceFadcode = getItem(STORAGE_KEYS.CTSTK0004);
  return `${envProvider.REACT_APP_BACK_OFFICE_URL}/${currentPage}?navigatedFromCT=true&fadcode=${deviceFadcode}`;
};
