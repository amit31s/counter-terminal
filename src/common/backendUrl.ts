/* istanbul ignore file */

import { featureFlags } from "@ct/constants";
import isElectron from "is-electron";
import { envProvider } from "./platformHelper";

const forgeRockRedirectUrl =
  envProvider.NODE_ENV === "development" || isElectron()
    ? envProvider.REACT_APP_FORGEROCK_COGNITO_REDIRECT_ELECTRON
    : envProvider.REACT_APP_FORGEROCK_COGNITO_REDIRECT_SIM;

// should contain backend urls for different environments to make switching easier
export const SERVER_ROOT: string = `https://api.spm-${envProvider.REACT_APP_ENV}.com` || "";

export const AWS_COGNITO_REGION: string = envProvider.REACT_APP_AWS_COGNITO_REGION || "";
export const AWS_COGNITO_USER_POOL_ID: string =
  envProvider.REACT_APP_AWS_COGNITO_USER_POOL_ID || "";
export const AWS_COGNITO_WEB_CLIENT_ID: string =
  envProvider.REACT_APP_AWS_COGNITO_WEB_CLIENT_ID || "";

export const FORGEROCK_COGNITO_REGION: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_REGION || "";
export const FORGEROCK_COGNITO_USER_POOL_ID: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_USER_POOL_ID || "";
export const FORGEROCK_COGNITO_WEB_CLIENT_ID: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_WEB_CLIENT_ID || "";
export const FORGEROCK_COGNITO_OAUTH_DOMAIN: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_OAUTH_DOMAIN || "";
export const FORGEROCK_COGNITO_S3_BUCKET: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_S3_BUCKET || "";
export const FORGEROCK_COGNITO_REDIRECT: string = forgeRockRedirectUrl || "";

const federationManagerUrlLocal = "http://localhost:4000/apps/federation-manager";
const federationManagerUrlDeployed =
  (envProvider.REACT_APP_USING_ELECTRON === "true"
    ? envProvider.REACT_APP_FEDERATION_MANAGER_URL_STANDALONE
    : envProvider.REACT_APP_FEDERATION_MANAGER_URL_SIMULATOR) || "";

export const FEDERATION_MANAGER_URL: string = featureFlags?.useLocalFederationManager
  ? federationManagerUrlLocal
  : federationManagerUrlDeployed;

export const POL_DEVICE_SERVER_HOST: string = envProvider.REACT_APP_POL_DEVICE_SERVER_HOST || "";
export const POL_DEVICE_SERVER_SIMULATED: boolean =
  envProvider.REACT_APP_USING_ELECTRON === "true"
    ? envProvider.REACT_APP_POL_DEVICE_SERVER_SIMULATED_ELECTRON === "true"
    : envProvider.REACT_APP_POL_DEVICE_SERVER_SIMULATED_SIMULATOR === "true";
export const CASH_TRANSFER_OUT_DISABLED: boolean =
  envProvider.REACT_APP_CASH_TRANSFER_OUT_DIASBLED === "true" || false;
export const POUCH_ACCEPTANCE_DISABLED: boolean =
  envProvider.REACT_APP_POUCH_ACCEPTANCE_DISABLED === "true" || false;
export const LOGOUT_CASHDRAWER_DISABLED: boolean =
  envProvider.REACT_APP_LOGOUT_CASHDRAWER_DISABLED === "true" || false;
export const POUCH_DISPATCH_DISABLED: boolean =
  envProvider.REACT_APP_POUCH_DISPATCH_DISABLED === "true" || false;
export const PMS_VISIBLE: boolean = envProvider.REACT_APP_PMS_VISIBLE === "true" || false;
export const QUANTITY_BUTTON_DISABLED: boolean =
  envProvider.REACT_APP_QUANTITY_BUTTON_DISABLED === "true" || false;
export const INITIALISE_PED_ON_LOGIN: boolean =
  envProvider.REACT_APP_INITIALISE_PED_ON_LOGIN === "true" || false;
// allows for local override of tpv for pin pad
export const POL_PED_TPV_OVERRIDE: string = envProvider.REACT_APP_POL_PED_TPV_OVERRIDE || "";
export const MSR_ENABLED: boolean = envProvider.REACT_APP_MSR_ENABLED === "true" || false;
export const REACT_APP_FORGEROCK_COGNITO_WEB_CLIENT_ID =
  envProvider.REACT_APP_FORGEROCK_COGNITO_WEB_CLIENT_ID;
