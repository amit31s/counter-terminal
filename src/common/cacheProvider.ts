import Cache from "@aws-amplify/cache";
import { CacheItemOptions as AmplifyCacheInterface } from "@aws-amplify/cache/src/types/Cache";

Cache.configure({
  itemMaxSize: 400000, // 400kb
});

type CacheValue = string | number | boolean | Record<string, unknown> | unknown[];

const setItem = (key: string, value: CacheValue, options?: AmplifyCacheInterface): void =>
  Cache.setItem(key, value, options);
const getItem = (key: string, options?: AmplifyCacheInterface): CacheValue | unknown =>
  Cache.getItem(key, options);
const removeItem = (key: string): void => Cache.removeItem(key);
const clear = (): void => Cache.clear();

export default Object.freeze({
  setItem,
  getItem,
  removeItem,
  clear,
});
