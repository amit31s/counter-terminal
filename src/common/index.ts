export * from "./backendUrl";
export * from "./backOfficeUrl";
export * from "./components";
export * from "./getProduct";
export * from "./helpers";
export * from "./hooks";
export * from "./selectors";
export * from "./state";
export * from "./theme";
