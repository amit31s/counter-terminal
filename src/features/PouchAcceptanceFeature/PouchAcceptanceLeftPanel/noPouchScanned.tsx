import { STRING_CONSTANTS } from "@ct/constants";
import FontFamily from "@ct/utils/Scaling/FontFamily";
import { Center, Text } from "native-base";

export const NoPouchScanned = () => {
  return (
    <Center flex={1}>
      <Text fontSize={28} fontFamily={FontFamily.FONT_NUNITO_BOLD}>
        {STRING_CONSTANTS.PouchAcceptance.scanOrEnterBarcode1}
      </Text>
      <Text fontSize={28} fontFamily={FontFamily.FONT_NUNITO_BOLD}>
        {STRING_CONSTANTS.PouchAcceptance.scanOrEnterBarcode2}
      </Text>
    </Center>
  );
};
