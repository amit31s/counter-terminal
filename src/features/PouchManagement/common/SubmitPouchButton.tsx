import { StyledButton } from "@ct/components";
import { BUTTON, SCREENS } from "@ct/constants";
import { Flex } from "native-base";
import { StyleSheet } from "react-native";

import { useNavigate } from "react-router-dom";

type SubmitPouchProps = {
  submit: () => void;
  disabled: boolean;
};

const styles = StyleSheet.create({
  base: {
    marginLeft: "24px",
  },
  position: { position: "absolute", bottom: 15 },
});

// Submit pouch buttons for pouch acceptance and dispatch
export const SubmitPouchButton = ({ submit, disabled }: SubmitPouchProps) => {
  const navigate = useNavigate();

  const cancel = () => {
    navigate(SCREENS.HOME, { state: { from: SCREENS.POUCH_ACCEPTANCE } });
  };

  return (
    <Flex direction="row" justifyContent={"center"} style={styles.position}>
      <StyledButton
        testID={"cancel"}
        label={BUTTON.CTBTN0006}
        type="secondary"
        size="slim"
        onPress={cancel}
      />
      <StyledButton
        testID={"confirm"}
        isDisabled={disabled}
        label={BUTTON.CTBTN0008}
        styles={styles.base}
        type="tertiary"
        size="slim"
        onPress={submit}
      />
    </Flex>
  );
};
