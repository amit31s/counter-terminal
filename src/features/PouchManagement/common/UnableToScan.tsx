import { STRING_CONSTANTS, TEXT } from "@ct/constants";
import FontFamily from "@ct/utils/Scaling/FontFamily";
import { Center, Text } from "native-base";

export const UnableToScan = () => {
  return (
    <Center testID="callBCS">
      <Text fontSize={24} fontWeight={600} fontFamily={FontFamily.FONT_NUNITO_REGULAR}>
        {STRING_CONSTANTS.messages.Unabletoscan}
      </Text>
      <Text fontSize={24} fontWeight={600} fontFamily={FontFamily.FONT_NUNITO_SEMI_BOLD}>
        {TEXT.CTTXT00040}
      </Text>
    </Center>
  );
};
