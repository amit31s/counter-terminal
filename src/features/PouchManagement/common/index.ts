export * from "./SubmitPouchButton";
export * from "./UnableToScan";
