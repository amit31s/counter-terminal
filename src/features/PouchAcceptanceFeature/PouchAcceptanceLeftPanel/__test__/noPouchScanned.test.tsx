import { render } from "@ct/common";
import { NoPouchScanned } from "../noPouchScanned";

describe("No pouch scanner", () => {
  it("snapshot no pouch scanner", async () => {
    const view = render(<NoPouchScanned />);
    expect(view.baseElement).toMatchSnapshot();
  });
});
