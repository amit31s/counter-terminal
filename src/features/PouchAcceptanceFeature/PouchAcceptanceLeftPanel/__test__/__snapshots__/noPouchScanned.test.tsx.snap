// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No pouch scanner snapshot no pouch scanner 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        class="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flex-13awgt0 r-justifyContent-1777fci"
      >
        <div
          class="css-text-901oao r-color-1njgeiw r-fontFamily-p6yezw r-fontSize-zz5t5d r-fontStyle-13uqrnb r-fontWeight-16dba41 r-letterSpacing-oxtfae r-lineHeight-dhbnww"
          dir="auto"
        >
          Scan or enter pouch barcode in
        </div>
        <div
          class="css-text-901oao r-color-1njgeiw r-fontFamily-p6yezw r-fontSize-zz5t5d r-fontStyle-13uqrnb r-fontWeight-16dba41 r-letterSpacing-oxtfae r-lineHeight-dhbnww"
          dir="auto"
        >
          the text box on the right.
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;
